core/jvm/src/main/scala/scalaz/zio/RTS.scala

```diff
@@ -473,10 +399,8 @@ private object RTS {
                                 if (curIo eq null) {
                                   result = value
                                 }
-                              case ExitResult.Terminated(ts) =>
-                                curIo = IO.terminate0(ts)
-                              case ExitResult.Failed(e, ts) =>
-                                curIo = IO.fail0(e, ts)
+                              case ExitResult.Failed(cause) =>
+                                curIo = cause.toIO
```

Comment:
At this point, `toIO` is probably too simple to exist, and this can be replaced by `IO.fail0(cause)`.